### PR TITLE
Remove 16× MSAA support due to driver bugs and low performance

### LINKED ISSUE #37917: Vulkan: Enabling 16× MSAA freezes the editor
**Godot version:** Git https://github.com/godotengine/godot/commit/3c1218a39a3cd4311812e7219d20b9cc735b4662

**OS/device including version:** Fedora 31, NVIDIA 440.64, GeForce GTX 1080

**Issue description:** Setting MSAA to 16× in the Project Settings will cause the Godot editor window to freeze, while also freezing the OS for 10+ seconds. The editor window must be killed before the OS becomes smooth again.

This doesn't occur with lower MSAA values like 2×, 4× and 8×.

See https://github.com/godotengine/godot/issues/14727#issuecomment-614225784 for an explanation of why this likely happens only with 16× MSAA.

**Steps to reproduce:** Set **Rendering > Quality > Screen Filters > Msaa** to **16x**. Note that the setting may be saved to the Project Settings if you're unlucky. If this happens, edit `project.godot` and remove the line `quality/screen_filters/msaa=4` to make the project editable again.

### PR DESCRIPTION
**Pull request by Calinou**
In the `master` branch, 16× MSAA caused the entire system to freeze on NVIDIA GPUs. This is likely caused by graphics drivers not actually implementing 16× MSAA, but combining 8× MSAA with 2× SSAA instead. If you need to implement a "photo mode" for high quality in-game screenshots, you can combine [2× SSAA](https://github.com/godotengine/godot-demo-projects/tree/master/viewport/3d_scaling) with 8× MSAA manually.

On top of that, modern shader complexity makes 16× MSAA very difficult to use while keeping a good framerate. This is especially true when you start using GIProbe or SDFGI. 8× MSAA is hard enough to use as it is, and it's not much of a loss visually compared to 16× MSAA.

In the `3.x` branch, there are only minor bugs with 16× MSAA such as https://github.com/godotengine/godot/issues/14727, so I think we can keep it there. Shaders are also not as complex in `3.x`, so higher MSAA values are more usable in practice.

This closes https://github.com/godotengine/godot/issues/37917.

### CODE REVIEW

File: scene/main/viewport.cpp [hunk 1]
```diff
@@ -3601,7 +3601,7 @@ void Viewport::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "snap_2d_transforms_to_pixel"), "set_snap_2d_transforms_to_pixel", "is_snap_2d_transforms_to_pixel_enabled");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "snap_2d_vertices_to_pixel"), "set_snap_2d_vertices_to_pixel", "is_snap_2d_vertices_to_pixel_enabled");
 	ADD_GROUP("Rendering", "");
-	ADD_PROPERTY(PropertyInfo(Variant::INT, "msaa", PROPERTY_HINT_ENUM, "Disabled (Fastest),2x (Fast),4x (Average),8x (Slow),16x (Slower)"), "set_msaa", "get_msaa");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "msaa", PROPERTY_HINT_ENUM, "Disabled (Fastest),2x (Average),4x (Slow),8x (Slower)"), "set_msaa", "get_msaa");
```
Comment (nathanfranke) [inline scene/main/viewport.cpp:3604]: I think "Moderate" is better than "Average"

Comment (Calinou) [reply scene/main/viewport.cpp:3604]: We use "Average" in other places, so I'd rather change them all in one go to be consistent.

Comment (reduz) [reply scene/main/viewport.cpp:3604]: Should it no be 8x (Slowest) ?

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**